File: python/Day11/factorial.py
# for printing a factorial

num=(int(input("Enter the number =  ")))
fact=1
if num<0:
    print("factorial does not exist for negative numbers")
elif num==0:
    print("the factorial of 0 is 1")
else:
    for i in range(1,num+1):
        fact=fact*i
    print("the factorial of",num,"is",fact)

'''



num = int(input("Enter the number = "))

def factorial(n):
    if n <= 0:
        print("factorial undefined")
        return 1
    result = 1
    for i in range(1, n + 1):
        result *= i
    return result  # ← Now properly indented inside the function

print("the factorial of", num, "is", factorial(num))


























'''




num=(int(input("Enter the number =  ")))
def factorial(n):
    if n<=0:
        print(" factorial undefined")
        return 1
    for i in range(1, n+ 1):
        return *= i
    return result


num=(int(input("Enter the number =  ")))

def factorial(n):
    return 1 if n<=1 else n* factorial(n-1)

File: python/factorial.java
package python;
import java.util.Scanner;
public class factorial {
    public static void main(String[] args) {
        System.out.println("Enter your number");
        int n;
        double fact = 1;
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        for(int i=1; i <=n;i++){
             fact = fact * i;
         System.out.println("i = " + i + ", fact = " + fact);
        }
        System.out.println(fact);
    }
}
